checkov/terraform/checks/resource/azure/AKSRbacEnabled.py

```diff
@@ -1,17 +1,37 @@
-from checkov.common.models.enums import CheckCategories
-from checkov.terraform.checks.resource.base_resource_value_check import BaseResourceValueCheck
+import dpath.util
+from checkov.common.models.enums import CheckCategories, CheckResult
+from checkov.terraform.checks.resource.base_resource_check import BaseResourceCheck
 
 
-class AKSRbacEnabled(BaseResourceValueCheck):
+class AKSRbacEnabled(BaseResourceCheck):
     def __init__(self):
         name = "Ensure RBAC is enabled on AKS clusters"
         id = "CKV_AZURE_5"
-        supported_resources = ['azurerm_kubernetes_cluster']
+        supported_resources = ["azurerm_kubernetes_cluster"]
         categories = [CheckCategories.KUBERNETES]
-        super().__init__(name=name, id=id, categories=categories, supported_resources=supported_resources)
+        super().__init__(
+            name=name,
+            id=id,
+            categories=categories,
+            supported_resources=supported_resources,
+        )
 
-    def get_inspected_key(self):
-        return 'role_based_access_control/[0]/enabled'
+    def scan_resource_conf(self, conf):
+        # azurerm < 2.99.0
+        inspected_key1 = "role_based_access_control/[0]/enabled"
+        # azurerm >= 2.99.0
+        inspected_key2 = "role_based_access_control_enabled"
+        rbac_enabled = False
+
+        if dpath.search(conf, inspected_key1) != {}:
+            rbac_enabled = dpath.get(conf, inspected_key1)[0]
+        elif dpath.search(conf, inspected_key2) != {}:
+            rbac_enabled = dpath.get(conf, inspected_key2)[0]
+
+        if rbac_enabled:
```

Comment:
Before returning the check results, please set the correct `self.evaluated_keys`, like it is done in this check: https://github.com/bridgecrewio/checkov/blob/7de0f84e320a82c36c7460edd4641d8ba3cf0fcc/checkov/terraform/checks/resource/aws/CloudfrontDistributionEncryption.py

Reply:
@rotemavni Thanks for the comment. I definitely overlooked the need for presence of this parameter.

I think changing the code to:

```python
        self.evaluated_keys = [
            "role_based_access_control/[0]/enabled",  # azurerm < 2.99.0
            "role_based_access_control_enabled",  # azurerm >= 2.99.0
        ]

        for key in self.evaluated_keys:
            if dpath.search(conf, key) and dpath.get(conf, key)[0]:
                return CheckResult.PASSED

        return CheckResult.FAILED
```

would make it more concise. The last commit reflects that change.